libraries/AP_TemperatureSensor/AP_TemperatureSensor.h

```diff
@@ -0,0 +1,123 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include <AP_HAL/AP_HAL.h>
+
+#ifdef AP_TEMPERATURE_SENSOR_ENABLED
+    #define AP_TEMPERATURE_SENSOR_ENABLED_BY_USER AP_TEMPERATURE_SENSOR_ENABLED
+#else
+    #define AP_TEMPERATURE_SENSOR_ENABLED_BY_USER 0
+    #define AP_TEMPERATURE_SENSOR_ENABLED (!HAL_MINIMIZE_FEATURES && BOARD_FLASH_SIZE > 1024)
+#endif
+
+#if AP_TEMPERATURE_SENSOR_ENABLED
+#include "AP_TemperatureSensor_Params.h"
+
+// maximum number of Temperature Sensors
```

Comment:
this should be in AP_TemperatureSensor_config.h, our new approach to the ifdefs checks

Reply:
done